wp-modules/editor/model.php

```diff
@@ -54,13 +54,14 @@ function save_pattern_to_file( WP_Post $post ) {
 		return;
 	}
 
-	$pattern = get_pattern_by_name( $post->post_name );
+	$name    = $post->post_name;
+	$pattern = get_pattern_by_name( $name );
 	update_pattern(
 		array_merge(
-			$pattern ? $pattern : [],
+			$pattern ? $pattern : [ 'slug' => $name ],
```

Comment:
This is the key change. 

Instead of setting the slug to the name in [pattern-data-handlers.php](wp-modules/pattern-data-handlers/pattern-data-handlers.php), this sets the slug to the name only on new patterns.

For patterns in Twenty Twentythree and Frost, they'll already have a slug that won't match the name.

We can respect the slug they set, and only change the slug when the Title (and therefore name) changes.

Reply:
This is such a simple way to check if the pattern is new — it's super clean!

Reply:
Wow, thanks!

Reply:
Yeah this is great. With this being so crucial to understand clearly, could we add some explainer inline comments?